### PR TITLE
id3v2lib: Add new recipe.

### PR DESCRIPTION
**Pull request by simoncent**
Specify library name and version:  **id3v2lib/2.1**

Add a recipe for https://github.com/larsbs/id3v2lib.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/id3v2lib/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def validate(self):
+        # An issue has been opened to discuss supporting MSVC:
+        # https://github.com/larsbs/id3v2lib/issues/48
+        if self.settings.compiler == "msvc" or str(self.settings.compiler) == "Visual Studio":
```
Comment (RubenRBS) [inline recipes/id3v2lib/all/conanfile.py:33]:
```suggestion
        if is_msvc(self):
```

File: recipes/id3v2lib/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get
```
Comment (RubenRBS) [inline recipes/id3v2lib/all/conanfile.py:3]:
```suggestion
from conan.tools.files import copy, get
from conan.tools.microsoft import is_msvc
```

File: recipes/id3v2lib/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+class LibwebmConan(ConanFile):
+    name = "id3v2lib"
+    description = "id3v2lib is a library written in C to read and edit id3 tags from mp3 files."
+    topics = ("conan", "id3", "tags", "mp3", "container", "media", "audio")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/larsbs/id3v2lib"
+    license = "BSD-2-Clause"
+
+    settings = "os", "arch", "compiler", "build_type"
```
Comment (RubenRBS) [inline recipes/id3v2lib/all/conanfile.py:18]:
```suggestion
    license = "BSD-2-Clause"
    package_type = "library"

    settings = "os", "arch", "compiler", "build_type"
```

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot for your contribution, we really appreciate it :)

File: recipes/id3v2lib/all/conanfile.py (hunk 1 above)
Comment (simoncent) [reply recipes/id3v2lib/all/conanfile.py]: Oh wow, how did I miss that? That's **much** better, thanks 👍🏻

File: recipes/id3v2lib/all/test_v1_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from conans import ConanFile, CMake, tools
```
Comment (uilianries) [inline recipes/id3v2lib/all/test_v1_package/conanfile.py:1]: The test_v1_package is no longer required for new recipes. You can remove it if you want: https://github.com/conan-io/conan-center-index/discussions/17406

**Review by uilianries — APPROVED** [review]
LGTM

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**